pytest_tests/testsuites/services/http_gate/test_http_headers.py

```diff
@@ -0,0 +1,205 @@
+
+        yield storage_objects
+
+    @allure.title("Get object1 by attribute")
+    def test_object1_can_be_get_by_attr(
+        self, storage_objects_with_attributes: list[StorageObjectInfo]
+    ):
+        """
+        Test to get object#1 by attribute and comapre hashes
+
+        Steps:
+        1. Download object#1 with attributes [Chapter2=w@r] and compare hashes
+        """
+
+        storage_object_1 = storage_objects_with_attributes[0]
+
+        with allure.step(
+            "Download object#1 via wget with attributes [Chapter2=w@r] and compare hashes"
+        ):
+            get_object_by_attr_and_verify_hashes(
+                oid=storage_object_1.oid,
+                file_name=storage_object_1.file_path,
+                cid=storage_object_1.cid,
+                attrs={"Chapter2": storage_object_1.attributes["Chapter2"]},
+                endpoint=self.cluster.default_http_gate_endpoint,
+            )
+
+    @allure.title("Test get object2 with different attributes, then delete object2 and get object1")
+    def test_object2_can_be_get_by_attr(
+        self, storage_objects_with_attributes: list[StorageObjectInfo]
+    ):
+        """
+        Test to get object2 with different attributes, then delete object2 and get object1 using 1st attribute. Note: obj1 and obj2 have the same attribute#1,
+        and when obj2 is deleted you can get obj1 by 1st attribute
+
+        Steps:
+        1. Download object#2 with attributes [chapter2=w@r] and compare hashes
+        2. Download object#2 with attributes [Ch@pter1=peace] and compare hashes
+        3. Delete object#2
+        4. Download object#1 with attributes [Writer=Leo Tolstoy] and compare hashes
+        """
+        storage_object_1 = storage_objects_with_attributes[0]
+        storage_object_2 = storage_objects_with_attributes[1]
+
+        with allure.step(
+            "Download object#2 via wget with attributes [chapter2=w@r] / [Ch@pter1=peace]  and compare hashes"
+        ):
+            selected_attributes_object2 = [
+                {"chapter2": storage_object_2.attributes["chapter2"]},
+                {"Ch@pter1": storage_object_2.attributes["Ch@pter1"]},
+            ]
+            for attributes in selected_attributes_object2:
+                get_object_by_attr_and_verify_hashes(
+                    oid=storage_object_2.oid,
+                    file_name=storage_object_2.file_path,
+                    cid=storage_object_2.cid,
+                    attrs=attributes,
+                    endpoint=self.cluster.default_http_gate_endpoint,
+                )
+        with allure.step("Delete object#2"):
+            delete_object(
+                wallet=self.wallet,
+                cid=storage_object_2.cid,
+                oid=storage_object_2.oid,
+                shell=self.shell,
+                endpoint=self.cluster.default_rpc_endpoint,
+            )
+            storage_objects_with_attributes.remove(storage_object_2)
```

Comment:
please add some check here. is this object deleted?

Reply:
done, thanks!